### ISSUE TITLE
[UG] Ambiguous parameters used for LIST

### ISSUE THREAD

**Issue by saajidshaik02** [saajidshaik02/pe#4]
![image.png](https://raw.githubusercontent.com/saajidshaik02/pe/main/files/ccc49970-8228-42fb-bf53-ae079cd17d92.png)

---

To me as a user, what is the difference between [all]... , [contacts]... and [allcontacts]...

I'm confused because the default "list" already shows all contacts, so putting these 3 variables here means?

Upon typing "help list I'm greeted with a (for a lack of better word) non-useful answer

![image.png](https://raw.githubusercontent.com/saajidshaik02/pe/main/files/81b97b8a-dc4e-494e-9cd4-4cf75c61118c.png)
---

**If there is a use** for the 3 of these optional parameters, **at least** it should be mentioned in the UG or under the LIST command so users like me aren't left stunned.

Hope this helps :0

**Comment by nus-pe-script** [comment]
# Team's Response
TLDR: read the last paragraph (but do read the entire thing to understand the context). <br>

Thanks for raising this up. The concerns of "what is the difference between [all]... , [contacts]... and [allcontacts]..." didn’t occur to us and while we do not think putting these 3 optional parameters are issues, interestingly not one but two raise the issue relating to the design of `list` command format, which set us thinking. Perhaps, we will start off with explaining why we have this design choice before addressing your concern at hand. <br>

In the AB3 product, the command for listing all persons is `list`. The `list` command is described as a command that does not take in parameters, but if there are any extraneous parameters, the command will accept them anyway and execute the command as if there weren’t any parameters. We describe as such based on the AB3 user guide: <br>

> Extraneous parameters for commands that do not take in parameters (such as `help`, `list`, `exit` and `clear`) will be ignored. <br>

**Similarly**, we wanted our `list` command to follow the same design principle as AB3 `list` command, but instead of just accepting _any_ extraneous parameters, our `list` command only accept the following extraneous parameters:
* `all`
* `contacts`
* `allcontacts`

Why such specificity, you may ask. One of us feels that the `list` and `find` command can be easily mixed up. Think about it, in our product, if we want to list all contacts that has the student status as PhD, we input `find ss/Phd`. But semantically, this is equivalent to saying `list Phd` or `list Phd contacts`. If `list` accepts any kind of extraneous parameters, it is not very helpful for the user that everytime the user inputs `list Phd`, all the contacts are still listed and the filter did not take place. Made worse, there isn’t any form of error message but a message that shows insistence to list all contacts. With that in mind, it may be better to just invalidate the command to inform the user from the get go that this is an error. <br>

While one may argue that the user should be intelligent enough to visit the user guide and read them to understand the product behaviour, we wanted to make the product more user friendly by invalidating list commands that are semantically similar to a find command. And our attempt for this is to have such a list command format, as you have experienced using. <br>

That is also why the extraneous parameters we accept are only the three parameters mentioned above -- only those that seem to suggest that the user wants to list all contacts, but probably cannot remember the exact exact command either. This is also the reason why upon entering `help list`, the description explaining `list` mentions that this ain’t `find`: `List to show all contacts, as opposed to find command.` <br>

Therefore, following the phrasing AB3 user guide uses, we hope you can see the `list` command as a command that does not take in parameters, but if there are any extraneous parameters, the command will accept only specific extraneous parameters (rather than any extraneous parameters). In other words, our `list` command is not a command that intentionally has parameters for a reason since these parameters are considered “extra”. Our `list` command is designed to accommodate some variations only by accepting only certain types of extra. <br>

In order to indicate in a simple manner what the `list` command format is, we write it as `list [all]... [contacts]... [allcontacts]...` . These parameters are optional as indicated by the square brackets, and if one decides to write these parameters many many times, that’s fine as well because it is likely that the user intends to list all contacts anyway, so the command will interpret them as just `list`. <br>

But…it seems that while the command format: `list [all]... [contacts]... [allcontacts]...` accurately explains the command format allowed, it contains an implied meaning that we did not realise, let alone realising that it will _mislead_ the user’s understanding -- that these parameters, despite it being indicated as optional, are there _intentionally for a specific function_  (?)... which is not the case, as explained above. Note that when we write the command format as such, we feel that indicating optional parameters does not mean that it must serve any purpose, it just mean that we generously allow these words to be entered even though they are not required (thus the word, optional) <br>

Given the benefit of doubt, perhaps the most ideal case would be that we write the list command format as `list`, but to add notes to it to explain clearly the variations allowed, regardless how lengthy our explanation may be. <br>

Based on this analysis, we diagnose the bug as a Documentation bug -- due to how our phrasing for command format seems to throw people off, despite that strictly in terms of **literal** meaning, the meaning conveyed is accurate. We believe that this bug’s severity is low because if a user does not over interpret information that is presented, users should have no problems using the product. After all, users can still list all the contacts with our `list` command. We ensured that what we claimed to be the command format `list [all]... [contacts]... [allcontacts]...` works as it is. We did not write the list command format wrongly either. This should not affect the normal operations and does not deserve a medium severity. <br>

Conveying the literal meaning of commands format accurately is a standard we abide by for writing the documentation. Writing command formats, or any text, in a way that considers the possible assumptions users may bring (which in this case, the mention of parameters seems to oddly mean that it serves a function) never crossed our minds. After some thought now, while it is valid, it also feels that such exceedingly high standards for writing documentation may be too much of an effort or time to fit into the short timeline we have in creating the product. There are many other more important things to do such as ensuring the function of each feature are implemented well. Therefore, we like to indicate this as response.NotInScope.

In summary, we feel that the real bug is the not-too-ideal phrasing of the list command format, as we believe this is the one that seemingly confuses users. There is indeed no difference between the command formats that included vs not included the accepted parameters, and these parameters indeed do not serve any specific function. The severity should be low, as all the functions are working in order and the point of confusion was due to the user's assumptions on how the format should be interpreted. (assumptions that those parameters mean some intentional function). Writing in a way that considers the user's assumptions requires a large amount of time as time is required to refine phrasing, thus it does not warrant the need to do it before v1.6 or v1.5. Hence, we give a response.NotInScope.

## The 'Original' Bug

[The team marked this bug as a duplicate of the following bug]

># No difference between the list contacts commands
>
>When doing the command `list` all the parameters `list all` or `list contacts` do the exact same thing and it allows for commands like `list all all contacts allcontacts` which does not make sense.
>
>I feel like there could be some explanation regarding what the difference between the different list commands is to make more sense or remove the functionality all together instead of misleading the user
>
>
>
>
>
><hr>
><sub>[original: nus-cs2103-AY2425S1/pe-interim#3131] [original labels: severity.Medium type.FeatureFlaw]</sub>
>
>
>

## Their Response to the 'Original' Bug

[This is the team's response to the above 'original' bug]

>TLDR: read the last paragraph (but do read the entire thing to understand the context). <br>
>
>Thanks for raising this up. These concerns didn’t occur to us and while we do not think these are issues, interestingly not one but two raise the issue relating to the design of `list` command format, which set us thinking. Perhaps, we will start off with explaining why we have this design choice before addressing your concern at hand. <br>
>
>In the AB3 product, the command for listing all persons is `list`. The `list` command is described as a command that does not take in parameters, but if there are any extraneous parameters, the command will accept them anyway and execute the command as if there weren’t any parameters. We describe as such based on the AB3 user guide: <br>
>
>> Extraneous parameters for commands that do not take in parameters (such as `help`, `list`, `exit` and `clear`) will be ignored. <br>
>
>**Similarly**, we wanted our `list` command to follow the same design principle as AB3 `list` command, but instead of just accepting _any_ extraneous parameters, our `list` command only accept the following extraneous parameters:
>* `all`
>* `contacts`
>* `allcontacts`
>
>Why such specificity, you may ask. One of us feels that the `list` and `find` command can be easily mixed up. Think about it, in our product, if we want to list all contacts that has the student status as PhD, we input `find ss/Phd`. But semantically, this is equivalent to saying `list Phd` or `list Phd contacts`. If `list` accepts any kind of extraneous parameters, it is not very helpful for the user that everytime the user inputs `list Phd`, all the contacts are still listed and the filter did not take place. Made worse, there isn’t any form of error message but a message that shows insistence to list all contacts. With that in mind, it may be better to just invalidate the command to inform the user from the get go that this is an error. <br>
>
>While one may argue that the user should be intelligent enough to visit the user guide and read them to understand the product behaviour, we wanted to make the product more user friendly by invalidating list commands that are semantically similar to a find command. And our attempt for this is to have such a list command format, as you have experienced using. <br>
>
>That is also why the extraneous parameters we accept are only the three parameters mentioned above -- only those that seem to suggest that the user wants to list all contacts, but probably cannot remember the exact exact command either. This is also the reason why upon entering `help list`, the description explaining `list` mentions that this ain’t `find`: `List to show all contacts, as opposed to find command.` <br>
>
>Therefore, following the phrasing AB3 user guide uses, we hope you can see the `list` command as a command that does not take in parameters, but if there are any extraneous parameters, the command will accept only specific extraneous parameters (rather than any extraneous parameters). In other words, our `list` command is not a command that intentionally has parameters for a reason since these parameters are considered “extra”. Our `list` command is designed to accommodate some variations only by accepting only certain types of extra. <br>
>
>In order to indicate in a simple manner what the `list` command format is, we write it as `list [all]... [contacts]... [allcontacts]...` . These parameters are optional as indicated by the square brackets, and if one decides to write these parameters many many times, that’s fine as well because it is likely that the user intends to list all contacts anyway, so the command will interpret them as just `list`. <br>
>
>But…it seems that while the command format: `list [all]... [contacts]... [allcontacts]...` accurately explains the command format allowed, it contains an implied meaning that we did not realise, let alone realising that it will _mislead_ the user’s understanding -- that these parameters, despite it being indicated as optional, are there _intentionally for a specific function_  (?)... which is not the case, as explained above. Note that when we write the command format as such, we feel that indicating optional parameters does not mean that it must serve any purpose, it just mean that we generously allow these words to be entered even though they are not required (thus the word, optional) <br>
>
>Given the benefit of doubt, perhaps the most ideal case would be that we write the list command format as `list`, but to add notes to it to explain clearly the variations allowed, regardless how lengthy our explanation may be. <br>
>
>Based on this analysis, we diagnose the bug as a Documentation bug -- due to how our phrasing for command format seems to throw people off, despite that strictly in terms of **literal** meaning, the meaning conveyed is accurate. We believe that this bug’s severity is low because if a user does not over interpret information that is presented, users should have no problems using the product. After all, users can still list all the contacts with our `list` command. We ensured that what we claimed to be the command format `list [all]... [contacts]... [allcontacts]...` works as it is. We did not write the list command format wrongly either. This should not affect the normal operations and does not deserve a medium severity. <br>
>
>Conveying the literal meaning of commands format accurately is a standard we abide by for writing the documentation. Writing command formats, or any text, in a way that considers the possible assumptions users may bring (which in this case, the mention of parameters seems to oddly mean that it serves a function) never crossed our minds. After some thought now, while it is valid, it also feels that such exceedingly high standards for writing documentation may be too much of an effort or time to fit into the short timeline we have in creating the product. There are many other more important things to do such as ensuring the function of each feature are implemented well. Therefore, we like to indicate this as response.NotInScope.
>
>In summary, we feel that the real bug is the not-too-ideal phrasing of the list command format, as we believe this is the one that seemingly confuses users. There is indeed no difference between the command formats that included vs not included the accepted parameters. The severity should be low, as all the functions are working in order and the point of confusion was due to the user's assumptions on how the format should be interpreted. (assumptions that those parameters mean some intentional function). Writing in a way that considers the user's assumptions requires a large amount of time as time is required to refine phrasing, thus it does not warrant the need to do it before v1.6 or v1.5. Hence, we give a response.NotInScope.
# Items for the Tester to Verify
## :question: Issue duplicate status

Team chose to mark this issue as a duplicate of another issue (as explained in the _**Team's response**_ above)

- [ ] I disagree

**Reason for disagreement:** [replace this with your explanation]

<catcher-end-of-segment><hr>
## :question: Issue response

Team chose [`response.NotInScope`]

- [x] I disagree

**Reason for disagreement:** Thank you for the detailed response. 

Firstly, the fact there are duplicate issues, is a prominent sign that its definitely not intuitive for users due to the missing information, especially since we testers also coincidentally fall under your target user group(students). 


Secondly, using "Exceedingly high standards for writing documentation"  and "writing in a way that consider's users assumption requires a large amount of time" as your logical arguments is a far stretch.


>![image.png](https://raw.githubusercontent.com/saajidshaik02/pe/main/files/29daf591-2026-4927-904d-c77b2d038100.png)

>![image.png](https://raw.githubusercontent.com/saajidshaik02/pe/main/files/2f9f8acd-4c91-44a4-82a0-a20d4aab3ba3.png)


---

Your arguments lies on the assumption that there already is **some sort of explanation** given to what each the optional parameters outputs to under `list` and that the amount of detail given is bare minimum and sufficient _(indicating that the user is expecting too much out of the UG)_. Unfortunately, there is **none at all**.  Let's take a look at your documentation.


![Screenshot 2024-11-15 164833.png](https://raw.githubusercontent.com/saajidshaik02/pe/main/files/db6dcc8b-c112-48b4-8a6c-00a7433dbb58.png)

Documentation is important, we don't expect you to write 2 or 3 pages explaining `list`, but based on the above screenshot, this is all that was present. 

>**How am I supposed to know that each of the optional parameters, `[all]` `[contacts]` and `[allcontacts]`, will give the same output? There are other optional parameters in your other commands, where upon their inclusion, will result in different outcomes.**
>
> **This odd behvaiour, that typing any of these different parameters, will return the same result, should be documented because it's not only a lack of detail, but also an easy point for confusion for users. This is different from your normal cosmetic bugs because there is equivalent to having no explanation on what the parts of the command does, which affects users heavily.** 


Additionally, to me, its a **contradiction** how something that you claim is to be **intutive** to understand and therefore not requiring explanation in UG, would end up being **super lengthy** to write in documentation. Even if I'm not a student under CS2103T, to a normal person, its intuitive that the documentation for this `list` command would not take more than half a page and can be simple to understand. 


And this is **perfectly supported by your very own response**. Your explanation in this response (to supplement the lack of information in the UG) is clear and sound. You even gave a detailed response here and it only took half a page (with other non-UG required information involved too).

Only by reading this reponse, I understand it better now. This is exactly what we were looking in the UG, which is also part of your requirements for the TP which failed to be present.

Evidently, **multiple times throughout your response**, you acknowledged there can be lapses in logical links and lack of documentation, many times, paired with suggestive words/**words indicating plausibility** such as "benefit of doubt", "perhaps" and etc. This is not convincing at all if you intend your audience to find that specific one way of thought process only, to figure out the parameters actually don't matter at all, without any guidance in the UG, showing there is indeed a legitimate issue.

**It is very very evident** that the command is confusing and there was an oversight in documentation, and this is a documentation bug that is within scope. 

---
Lastly, lets consider what response is considered notInScope. You mentioned below, "Exceedingly high standards for writing documentation" may be too much time invested when you could be improving other aspects of the product. Let's take a look at the rubrics.

![Screenshot 2024-11-18 221707.png](https://raw.githubusercontent.com/saajidshaik02/pe/main/files/aa577a40-8d8e-406b-9116-2e43d9a1fa38.png)

It's neither a missing feature, a suboptimal design of feature _(its a documentation bug not a feature, even then, its completely missing documentation, not unclear documentation!)_ nor a known bug. It does not satisfy any of the bulletpoints. Lastly, as mentioned above, this is not something that would takes excessive time to the point it hinders production!

There are no grounds to be placing this under `response.notInResponse`. Hope this clarifies my response. Thank you!

<catcher-end-of-segment><hr>
